crates/nu-command/src/filters/values.rs

```diff
@@ -0,0 +1,203 @@
+                        Value::test_int(32),
+                    ],
+                    span: Span::test_data(),
+                }),
+            },
+            Example {
+                example: "[[name meaning]; [ls list] [mv move] [cd 'change directory']] | values",
+                description: "Get the values from the table (produce a list of lists)",
+                result: Some(Value::List {
+                    vals: vec![
+                        Value::List {
+                            vals: vec![
+                                Value::test_string("ls"),
+                                Value::test_string("mv"),
+                                Value::test_string("cd"),
+                            ],
+                            span: Span::test_data(),
+                        },
+                        Value::List {
+                            vals: vec![
+                                Value::test_string("list"),
+                                Value::test_string("move"),
+                                Value::test_string("change directory"),
+                            ],
+                            span: Span::test_data(),
+                        },
+                    ],
+                    span: Span::test_data(),
+                }),
+            },
+        ]
+    }
+
+    fn run(
+        &self,
+        engine_state: &EngineState,
+        _stack: &mut Stack,
+        call: &Call,
+        input: PipelineData,
+    ) -> Result<PipelineData, ShellError> {
+        let span = call.head;
+        values(engine_state, span, input)
+    }
+}
+
+// The semantics of `values` are as follows:
+// For each column, get the values for that column, in row order.
+// Holes are not preserved, i.e. position in the resulting list
+// does not necessarily equal row number.
+pub fn get_values<'a>(
+    input: impl IntoIterator<Item = &'a Value>,
+    head: Span,
+    input_span: Span,
+) -> Result<Vec<Value>, ShellError> {
+    let mut output: IndexMap<String, Vec<Value>> = IndexMap::new();
+
+    for item in input {
+        match item {
+            Value::Record { cols, vals, .. } => {
+                for (k, v) in cols.iter().zip(vals.iter()) {
+                    if let Some(vec) = output.get_mut(k) {
+                        vec.push(v.clone());
+                    } else {
+                        output.insert(k.clone(), vec![v.clone()]);
+                    }
+                }
+            }
+            Value::Error { error } => return Err(error.clone()),
+            _ => {
+                return Err(ShellError::OnlySupportsThisInputType(
+                    "record or table".into(),
+                    item.get_type().to_string(),
+                    head,
+                    input_span,
+                ))
+            }
+        }
+    }
+
+    Ok(output.into_values().map(|v| Value::list(v, head)).collect())
+}
+
+fn values(
+    engine_state: &EngineState,
+    head: Span,
+    input: PipelineData,
```

Comment:
It would be nice to pass through the pipeline metadata so we could have ls_colors if one did something like `(ls | values).0`.  Could you please investigate that? I think you could do something like this:
```rust
let metadata = input.metadata();
...
Ok(blah.into_pipeline_data(ctrlc).set_metadata(metadata))
```

Reply:
Idea: shouldn't `into_pipeline_data()` *require* the metadata be passed into it, similar to Values and spans? That way, this sort of issue wouldn't have to come up.

Reply:
Anyway, I've added the code you asked for. Note that `(ls | values).0` will still not work with it because the code for showing ls_colors in viewers/table.rs [does not run at all unless the data is a Value::Record with a 'name' column](https://github.com/nushell/nushell/blob/main/crates/nu-command/src/viewers/table.rs#L638) 🤸‍♀️ so it won't have any visible difference.

Reply:
> Idea: shouldn't into_pipeline_data() require the metadata be passed into it, similar to Values and spans? That way, this sort of issue wouldn't have to come up.

Probably? Maybe? I know we've had the same problem for a long time where metadata wasn't passed into a variety of commands.

> [does not run at all unless the data is a Value::Record with a 'name' column](https://github.com/nushell/nushell/blob/main/crates/nu-command/src/viewers/table.rs#L619-L656)

oof, I forgot about that, and I wrote it. LOL